code/registration.py

```diff
@@ -1236,51 +1363,32 @@ def singleplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
         path to h5 file
     output_dir: Path
         output directory
-    frame_rate_hz: float
-        frame rate in Hz
+    reference_image_fp: Path
+        path to reference image
     """
-    try:
-        h5_file = next(datainput.glob("*.h5"))
-    except:
-        h5_file = next(datainput.glob("*/*/*.h5"))
-
-    session_fp = h5_file.parent / "session.json"
-    with open(session_fp, "r") as j:
-        session_data = json.load(j)
-    frame_rate_hz = session_data["data_streams"][0]["ophys_fovs"][0]["frame_rate"]
-    experiment_id = "bergamo"
-    output_dir = make_output_directory(output_dir, experiment_id)
-    good_epochs = ["spont", "pair_neuron6_and_7_10xmult", "pair_neuron6_and_7"]
-    output_h5_file = Path(output_dir) / "bergamo.h5"
+    if not h5_file.is_file():
+        h5_file = [f for f in h5_file.rglob("*.h5") if unique_id in str(f)][0]
+    print(f"Running h5 file: {h5_file}")
+    output_dir = make_output_directory(output_dir, unique_id)
+    reference_image_fp = generate_bergamo_movies(h5_file, session)
+    if debug:
+        stem = h5_file.stem
+        debug_file = Path("../scratch") / f"{stem}_debug.h5"
+        with h5py.File(h5_file, "r") as f:
```

Comment:
Fine, though 5000 seems still quite a lot of frames for debugging purposes, and we could have used a virtual dataset.

Reply:
Adding this as an issue